packages/legend-query/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -69,7 +69,19 @@ const DerivedPropertyParameterEditor = observer(
           item: QueryBuilderParameterDragSource,
           monitor: DropTargetMonitor,
         ): void => {
-          if (!monitor.didDrop()) {
+          if (
+            !monitor.didDrop() &&
+            (derivedPropertyExpressionState.parameters[
```

Comment:
this code is kind of hard to read. Could we maybe abstract this out and name it exactly what we are doing. We checking compatiblity of types right?